nova_vm/src/ecmascript/builtins/numbers_and_dates/math_object.rs

```diff
@@ -342,122 +341,687 @@ impl MathObject {
-        let _x = to_number(agent, arguments.get(0))?;
-        todo!();
+        // 1. Let n be ? ToNumber(x).
+        let n = to_number(agent, arguments.get(0))?;
+
+        // 2. If n is either NaN or +∞𝔽, return n.
+        if n.is_nan(agent) || n.is_pos_infinity(agent) {
+            return Ok(n.into_value());
+        }
+
+        let n = n.into_f64(agent);
+
+        // 3. If n is 1𝔽, return +0𝔽.
+        if n == 1.0 {
+            return Ok(Value::pos_zero());
+        }
+
+        // 4. If n < 1𝔽, return NaN.
+        if n < 1.0 {
+            return Ok(Value::nan());
+        }
+
+        // 5. Return an implementation-approximated Number value representing the result of the inverse hyperbolic cosine of ℝ(n).
+        Ok(Value::from_f64(agent, n.acosh()))
     }
 
     fn asin(agent: &mut Agent, _this_value: Value, arguments: ArgumentsList) -> JsResult<Value> {
-        let _x = to_number(agent, arguments.get(0))?;
-        todo!();
+        // 1. Let n be ? ToNumber(x).
+        let n = to_number(agent, arguments.get(0))?;
+
+        // 2. If n is one of NaN, +0𝔽, or -0𝔽, return n.
+        if n.is_nan(agent) || n.is_pos_zero(agent) || n.is_neg_zero(agent) {
+            return Ok(n.into_value());
+        }
+
+        let n = n.into_f64(agent);
+
+        // 3. If n > 1𝔽 or n < -1𝔽, return NaN.
+        if !(-1.0..=1.0).contains(&n) {
+            return Ok(Value::nan());
+        }
+
+        // 4. Return an implementation-approximated Number value representing the result of the inverse sine of ℝ(n).
+        Ok(Value::from_f64(agent, n.asin()))
     }
 
     fn asinh(agent: &mut Agent, _this_value: Value, arguments: ArgumentsList) -> JsResult<Value> {
-        let _x = to_number(agent, arguments.get(0))?;
-        todo!();
+        // 1. Let n be ? ToNumber(x).
+        let n = to_number(agent, arguments.get(0))?;
+
+        // 2. If n is not finite or n is either +0𝔽 or -0𝔽, return n.
+        if !n.is_finite(agent) || n.is_pos_zero(agent) || n.is_neg_zero(agent) {
+            return Ok(n.into_value());
+        }
+
+        // 3. Return an implementation-approximated Number value representing the result of the inverse hyperbolic sine of ℝ(n).
+        Ok(Value::from_f64(agent, n.into_f64(agent).asinh()))
     }
 
     fn atan(agent: &mut Agent, _this_value: Value, arguments: ArgumentsList) -> JsResult<Value> {
-        let _x = to_number(agent, arguments.get(0))?;
-        todo!();
+        // 1. Let n be ? ToNumber(x).
+        let n = to_number(agent, arguments.get(0))?;
+
+        // 2. If n is one of NaN, +0𝔽, or -0𝔽, return n.
+        if n.is_nan(agent) || n.is_pos_zero(agent) || n.is_neg_zero(agent) {
+            return Ok(n.into_value());
+        }
+
+        // 3. If n is +∞𝔽, return an implementation-approximated Number value representing π / 2.
+        if n.is_pos_infinity(agent) {
+            return Ok(Value::from_f64(agent, consts::FRAC_PI_2));
```

Comment:
Something like the `BUILTIN_STRINGS` but for numbers? It would make sense but feels like it is probably a bit out of scope for this pr. Should I create an issue for it instead?